docs/source/models.rst

```diff
@@ -1122,6 +1122,194 @@ in native spaCy format.
 
 For more information, see :py:mod:`mlflow.spacy`.
 
+``Spacy`` pyfunc usage
+~~~~~~~~~~~~~~~~~~~~~~
+
+The example below shows how to train a ``Spacy`` ``TextCategorizer`` model, log the model artifact and metrics to the
+mlflow tracking server and then load the saved model to make predictions. For this example, we will be using the
+``Polarity 2.0`` dataset available in the ``nltk`` package. This dataset consists of 10000 positive and 10000 negative
+short movie reviews.
+
+First we convert the texts and sentiment labels ("pos" or "neg") from NLTK native format to ``Spacy``'s ``DocBin`` format:
+
+.. code-block:: python
```

Comment:
Hello! I understand that this example is a bit longer than most others. IMO, this is because Spacy has moved to a largely command line + config driven user interface compared to a programmatic user interface like pytorch and tensorflow. So the only way to have a single code script that does all of the things you mention would be to copy & paste part of Spacy's interface. Please see here: https://spacy.io/usage/training#quickstart

The only way I can think of making the example simpler is to not use a Text Classification problem, e.g. https://github.com/arnaudsj/sample-mlflow-spacy/blob/master/train_predict.py#L28-L39. This will remove the need for the data processing functions. I'm happy to do that, but IMO the text classification problem is something that we see in practice the most, so it's definite more useful for mlflow users. Please LMK which one you prefer.

Reply:
@dipanjank Understood! Didn't know Spacy has moved to a largely command line + config driven user interface. Given that, the current example looks reasonable.

Reply:
Cool, then I will fix the existing review comments in this PR.